lxd-agent/network.go

```diff
@@ -69,3 +77,89 @@ func serverTLSConfig() (*tls.Config, error) {
 
 	return tlsConfig, nil
 }
+
+// reconfigureNetworkInterfaces checks for the existence of "nics.json" in the config share.
+// Interfaces are identified by their MAC address and configured accordingly.
+func reconfigureNetworkInterfaces() {
+	// Abort if configuration file does not exist. (Nothing to do).
+	if !shared.PathExists("nics.json") {
+		return
+	}
+
+	nicBytes, err := ioutil.ReadFile("nics.json")
+	if err != nil {
+		return
+	}
+
+	nicData := make(map[string]instance.NICConfig)
+	err = json.Unmarshal(nicBytes, &nicData)
+	if err != nil {
+		return
+	}
+
+	netDir := "/sys/class/net"
+	files, err := ioutil.ReadDir(netDir)
+	if err != nil {
+		return
+	}
+
+	reverter := revert.New()
+	for _, f := range files {
+		addrBytes, err := ioutil.ReadFile(filepath.Join(netDir, f.Name(), "address"))
+		if err != nil {
+			return
+		}
+		macAddress := string(bytes.TrimRight(addrBytes, "\n"))
+
+		MTUBytes, err := ioutil.ReadFile(filepath.Join(netDir, f.Name(), "mtu"))
```

Comment:
Just a nit, and I realise it goes somewhat counter to my earlier comment, but when a variable isn't exported (and there's no need to here), we tend to use lowercase, so `mtuBytes`. :)